code/datums/components/item_killsound.dm

```diff
@@ -0,0 +1,39 @@
+/datum/component/item_killsound
+	/// list of allowed types, not null/empty
+	var/allowed_mobs
+	/// list of blacklisted types
+	var/blacklisted_mobs
+	var/killsound
+	var/killsound_volume = 100
+	/// on true will act as replacement for mob's death sound,
+	/// otherwise it will just play sound on death
```

Comment:
```suggestion
	/**
	 * on true will act as replacement for mob's death sound,
	 * otherwise it will just play sound on death
	 */
```
multiline docs